list.c

```diff
@@ -52,20 +52,34 @@ unsigned count_nodes(Head* head)
     return count;
 }
 
-void append_to_list(Head* head, unsigned size, void* data) // TODO: how would I know if my operation fails? return type should be bool
+int append_to_list(Head* head, unsigned size, void* data)
 {
-    if(head) {
-        if(!(head->front)) {
+    if (head) 
+    {
+        if (!(head->front)) 
+        {
             head->front = create_node();
             fill_node_data(head->front, size, data);
-        } else {
-            Node* this_node  = head->front;
-            Node* next_node = get_next_node(this_node);
-            while(next_node) {
+        } 
+        else 
+        {
+            Node *this_node = head->front;
+            Node *next_node = get_next_node(this_node);
+            while (next_node) 
+            {
                 this_node = next_node;
                 next_node = get_next_node(next_node);
             }
-            append_node(this_node, size, data);
+            if (append_node(this_node, size, data)) 
+            {
+                return 1;
+            } 
+            else 
+            {
+                return 0;
+            }
         }
+        return 1;

```

Comment:
**suggestion** (non-blocking): This return will never be reached, since the code will branch into one of `if`-`else` branches.

Reply:
> suggestion (non-blocking): This return will never be reached, since the code will branch into one of if-else branches.

Yes you are right- to improve the code and eliminate the unreachable `return 1` statement, we can simplify this logic by removing the need for an `else` block.